submissions/homework_8/cc1759/hw.sql

```diff
@@ -0,0 +1,261 @@
+--     WHERE authors.name = "Timotello";
+
+-- -- book isbn ---> book ---> book data
+-- EXPLAIN
+-- SELECT books.title, authors.name, books.year_published
+--     FROM books
+--     JOIN book_authors ON books.id = book_authors.book_id
+--     JOIN authors ON book_authors.author_id = authors.id
+--     WHERE books.isbn = 8781924907302;
+
+
+-- Composite Index
+
+CREATE INDEX idx_memberID_checkoutDate ON checkouts (book_id, checkout_date);
+CREATE INDEX idx_bookID_checkoutDate ON checkouts (member_id, checkout_date);
+-- -- member name ---> member ---> member data (their check outs, name, and phone#)
+-- EXPLAIN
+-- SELECT members.full_name AS name, members.phone_numb, books.title, checkouts.checkout_date
+--     FROM members
+--     JOIN checkouts ON members.id = checkouts.member_id
+--     JOIN books ON checkouts.book_id = books.id
+--     WHERE members.card_numb = "2fj3e9KnU0pkJnH"
+--     ORDER BY checkouts.checkout_date DESC;
+
+CREATE INDEX idx_title_yearPublished ON books (year_published, title);
+-- year published ---> book ---> book data
+-- EXPLAIN
+-- SELECT books.title, authors.name, books.year_published
+--     FROM books
+--     JOIN book_authors ON books.id = book_authors.book_id
+--     JOIN authors ON book_authors.author_id = authors.id
+--     WHERE year_published > 2010 AND year_published < 2020
+--     ORDER BY books.year_published ASC, books.title ASC;
+
+
+-- FUNCTION INDEX
+-- email ---> member profile ---> member data
+CREATE INDEX idx_email_lower ON web_profiles ((LOWER(email)));
+-- EXPLAIN
+-- SELECT members.full_name AS member, members.phone_numb, books.title, checkouts.checkout_date
+--     FROM web_profiles
+--     JOIN members ON web_profiles.member_id = members.id
+--     JOIN checkouts ON members.id = checkouts.member_id
+--     JOIN books ON checkouts.book_id = books.id
+--     WHERE LOWER(email) = "uncleweaven@gmail.com";
+
+
+-- Fulltext Index
+
+CREATE FULLTEXT INDEX idx_reviews ON book_reviews (review);
+-- book ---> positive reviews ---> review data (stars, web_profile, review, review date)
+-- EXPLAIN
+-- SELECT books.title, book_reviews.rating, members.full_name, 
+--     book_reviews.review, book_reviews.review_date
+--     FROM books
+--     JOIN book_reviews ON books.id = book_reviews.book_id
+--     JOIN web_profiles ON book_reviews.memb_web_prof_id =  web_profiles.id
+--     JOIN members ON web_profiles.member_id = members.id
+--     WHERE books.title = "The Adventures of Bobrickabilly"
+--     AND book_reviews.rating >= 4
+--     AND (book_reviews.review LIKE "awsome" 
+--         OR book_reviews.review LIKE "%great%" 
+--         OR book_reviews.review LIKE "%stupendis%" 
+--         OR book_reviews.review LIKE "%good%" 
+--         OR book_reviews.review LIKE "%love%" 
+--         OR book_reviews.review LIKE "%adore%" 
+--         OR book_reviews.review LIKE "%best%");
+
+
+-- FORIGN KEY INDEXES
+
+CREATE INDEX idx_ba_book_id ON book_authors (book_id);
```

Comment:
This index is redundant due to the composite index starting with `book_id` on this table already. Remember in a composite index, the index on the first column mentioned would be the same as an individual index on that first column. This is why the order matters.